pydantic/type_adapter.py

```diff
@@ -311,11 +311,11 @@ def core_schema(self) -> CoreSchema:
 
     @cached_property
     @_frame_depth(2)  # +2 for @cached_property + validator(self)
-    def validator(self) -> SchemaValidator:
+    def validator(self) -> SchemaValidator | PluggableSchemaValidator:
         """The pydantic-core SchemaValidator used to validate instances of the model."""
-        if not isinstance(self._validator, SchemaValidator):
+        if not isinstance(self._validator, (SchemaValidator, PluggableSchemaValidator)):
             self._init_core_attrs(rebuild_mocks=True)  # Do not expose MockValSer from public function
-        assert isinstance(self._validator, SchemaValidator)
+        assert isinstance(self._validator, (SchemaValidator, PluggableSchemaValidator))
```

Comment:
This is where the problem is — because we were asserting that it was a SchemaValidator, but it could be a PluggableSchemaValidator. While we could just drop this assert, @adriangb wanted to just stop lying in the `create_schema_validator` return type annotation.

Reply:
(And I agree with that choice, given I wasn't sure where else there might be problems.)

Ideally we would make it so that a PluggableSchemaValidator was actually a subclass of SchemaValidator, or similar, but anyway this works for now.